docker-compose.yml

```diff
@@ -62,7 +62,7 @@ services:
     build: .
     networks:
       - main
-    command: celery -A celerywyrm worker -l info -Q high_priority,medium_priority,low_priority,imports
+    command: celery -A celerywyrm worker -l info -Q high_priority,medium_priority,low_priority,imports,broadcast
```

Comment:
Sorry to chime in here unannounced. I haven't worked with Celery in a long time so I was curious about the ordering of this. I thought the `-Q` argument might define the priority in which the worker should be picking stuff from the queues. But apparently [according to the docs](https://docs.celeryq.dev/en/stable/getting-started/next-steps.html#routing) the order doesn't matter:
> The order of the queues doesn’t matter as the worker will give equal weight to the queues

So depending on how you run the worker processes on b.s you might just get one massive FIFO-style queueing system, but with multiple queue names.

Reply:
No need to apologize, chiming in is ideal. And yeah, the priority isn't meaningful on a technical level (I think there's a way to do this with rabbitmq but not redis? but that's a tangent). Celery seems to basically pick tasks off the top of one queue, then the top of another queue, et cetera, so the effect isn't that the high priority task queue is picked more often, but rather that it's a short queue, and when 10,000 import tasks are added, they aren't blocking a high priority task.

This approach of adding a queue did fix the problem with imports taking untenably long times to start, which is why I'm hopeful that it will improve the situation with posts not reaching remote servers in a timely manner (obviously only for posts created after this change, existing posts wouldn't be re-queued)

Reply:
So in a way, we are effectively getting the same benefit as if we were to scrap all those queues and just use one plus [Celery's message priorities](https://docs.celeryq.dev/en/stable/userguide/routing.html#redis-message-priorities)?

After reading more about scaling Celery, it seems that the way it handles queues and priorities, we would be better off running multiple workers with different `-Q` configurations, which would require each admin to tailor their setup better to the demand. i.e. in b.s's case with loads and loads of activity, having two separate worker configurations, one only on high-priority tasks, and the other on all or all other queues could help bring those high-priority queue lengths down a bit?

Reply:
Ah, there's also this config setting: https://docs.celeryq.dev/projects/kombu/en/latest/reference/kombu.transport.redis.html#kombu.transport.redis.Channel.queue_order_strategy

We would probably need to think this through properly, but using something other than the default `round_robin` might be worth trying out. Seems to me that by setting this to `priority` then the order of queues in the `-Q` would effectively be a priority list. But there are probably noticeable side-effects, lower priority queues would simply not get any tasks taken off of it if there aren't enough workers running. Instead of with round robin the whole system would just get slower under the same situation.

Reply:
I think that's worth trying out (both because it sounds like a better, in-built approach to the problem I'm trying to solve, and it would be less maintenance, especially across docker and non-docker), and I think I also misunderstood those docs the first time I read them.

I'm a little unsure how much work it would be to switch systems, however, and I'm tempted to keep using this approach in this PR and re-think it as future work. Do you have thoughts on that? I might be making it out to be a bigger task in my head that it really is.

Reply:
Yeah it makes sense to give this one a go on b.s and see how it behaves.

I don't think the code changes to switch from `round_robin` to `priority` or `ordered` are going to be that big, but probably makes sense to think it through and do a bit of testing before rolling it out 👍

Reply:
I'm guessing this has already been deployed to b.s? I'm seeing updates from that instance federate within at least 17 minutes. It's likely that it's even faster, just that my timing looking at my feed was off :) It used to take a least a day or two for b.s status updates to populate in my instance's feed.

Reply:
Yes, I'm using it there and federation has gotten a lot better, at least for outgoing activities.